### PR TITLE
#167165079 Implements Super Admin Functionality

### PR DESCRIPTION
**Pull request by OdunayoOkebunmi**
### What does this PR do?
a super admin can delete a user
can update a user details
can get all users
can create a user
[Starts #167165079]
#### Description of Task to be completed?
As a Super admin User
I want to be able to CREATE users and READ/ACCESS all users.
I want to be able to UPDATE and DELETE users.
I want to be able to CHANGE users' access level(type)
#### How should this be manually tested?
```bash
# fetch this branch
$ git fetch origin  ft-super-admin-167165079

# install dependencies
$ npm install

# run unit tests
$ npm test

# start the development server 
# make a post request to /admin/users to create a user
# make a patch request to /admin/users/:id to update a user
# make a delete request to /admin/users/:id to delete a user
# make a get request to /admin/users to get a user
$ npm run server
```
#### Any background context you want to provide?
API Flowchart

#### What are the relevant pivotal tracker stories?
- [#167165079](https://www.pivotaltracker.com/story/show/167165079)
#### Screenshots (if appropriate)
Screenshot 2019-08-08 at 18 21 16
Screenshot 2019-08-08 at 18 19 00
Screenshot 2019-08-09 at 15 31 37
Screenshot 2019-08-09 at 13 40 55

### CODE REVIEW

File: __tests__/admin.test.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      .delete(`${url}/${userId}`)
+      .set('Authorization', `Bearer ${verifiedUserToken}`)
+      .end((err, res) => {
+        expect(res.status).to.eql(200);
+        expect(res.body.user).to.be.a('object');
+        expect(res.body.user.message).to.eql('User has been deleted');
+        done();
+      });
+  });
+  it('should throw an error if user does has been deleted already', (done) => {
```
Comment (VictorMaria) [inline __tests__/admin.test.js]: Please review the text in line 97, Good job with testing edges cases.

**Review by VictorMaria — APPROVED** [review]

**Review by ocranbillions — APPROVED** [review]
LGTM

File: __tests__/admin.test.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  it('should successfully get all users', (done) => {
+    chai
+      .request(app)
+      .get(`${url}`)
+      .set('Authorization', `Bearer ${verifiedUserToken}`)
+      .end((err, res) => {
+        expect(res.status).to.eql(200);
+        expect(res.body.users[0]).to.have.property('id');
+        expect(res.body.users[0]).to.have.property('firstName');
+        expect(res.body.users[0]).to.have.property('lastName');
```
Comment (mazma1) [inline __tests__/admin.test.js:39]: No specific assertion here?

**Comment by mazma1** [comment]
@OdunayoOkebunmi did you update your screenshots to reflect the change(s)?

Can you also update your signup response object since we agreed to return only a user's token?

File: __tests__/admin.test.js (hunk 2 above)
Comment (OdunayoOkebunmi) [reply __tests__/admin.test.js:39]: > No specific assertion here?

No. that test suite is to get all the users

**Comment by OdunayoOkebunmi** [comment]
> @OdunayoOkebunmi did you update your screenshots to reflect the change(s)?
> 
> Can you also update your signup response object since we agreed to return only a user's token?

I have updated it

**Pull request merged**